From 0bb608f1b79e08c3959a663b280de1881ce7e62d Mon Sep 17 00:00:00 2001
From: Andy Hsu <andy.hsu.wiwynn@gmail.com>
Date: Tue, 21 Jul 2026 02:33:15 +0000
Subject: [PATCH] configuration: yosemite4: Add config for Wailua Falls

### Summary:
Related JIRA ticket:
https://metainfra.atlassian.net/jira/software/c/projects/YV4T1M/list?jql=project%20%3D%20YV4T1M%20ORDER%20BY%20created%20DESC&selectedIssue=YV4T1M-2363

# Description:
Add a configuration file for the Wailua Falls without ASIC.
Update the probe conditions in the Wailua Falls configuration file to enforce an exact match for the "Wailua Falls".

# Motivation:
A new configuration file is required to support the additional Wailua falls boards with the ASIC removed.

# Test Plan:
(1)Build: pass
(2)Test on Yv4: pass
---
 configurations/meson.build                    |   1 +
 .../meta/yv4/yosemite4_wailuafalls.json       |  49 ++++++++-
 .../yosemite4_wailuafalls_without_asic.json   | 102 ++++++++++++++++++
 3 files changed, 150 insertions(+), 2 deletions(-)
 create mode 100644 configurations/meta/yv4/yosemite4_wailuafalls_without_asic.json

diff --git a/configurations/meson.build b/configurations/meson.build
index 8c08062..e8ff3de 100644
--- a/configurations/meson.build
+++ b/configurations/meson.build
@@ -281,6 +281,7 @@ configs = [
     'meta/yv4/yosemite4_sentineldome_t2.json',
     'meta/yv4/yosemite4_sentineldome_t2_retimer.json',
     'meta/yv4/yosemite4_wailuafalls.json',
+    'meta/yv4/yosemite4_wailuafalls_without_asic.json',
     'meta/yv4/yosemite4n.json',
     'meta/yv5/yosemite5_1kw_paddle_board_hsc_mps.json',
     'meta/yv5/yosemite5_1kw_paddle_board_hsc_rds.json',
diff --git a/configurations/meta/yv4/yosemite4_wailuafalls.json b/configurations/meta/yv4/yosemite4_wailuafalls.json
index 99746ef..90fda8f 100644
--- a/configurations/meta/yv4/yosemite4_wailuafalls.json
+++ b/configurations/meta/yv4/yosemite4_wailuafalls.json
@@ -1,5 +1,50 @@
 {
     "Exposes": [
+        {
+            "Address": [
+                "7",
+                "236",
+                "128",
+                "1",
+                "0",
+                "$bus % 16 * 5"
+            ],
+            "Bus": "$bus / 4 - 4",
+            "Class": "I3C",
+            "EndpointId": "$bus % 15 * 10 + 2",
+            "Name": "BIC",
+            "Type": "MCTPEndpoint"
+        },
+        {
+            "Address": [
+                "7",
+                "236",
+                "128",
+                "1",
+                "0",
+                "$bus % 16 * 5"
+            ],
+            "Bus": "$bus / 4 - 4",
+            "Class": "I3C",
+            "EndpointId": "$bus % 15 * 10 + 4",
+            "Name": "CXL1",
+            "Type": "MCTPEndpoint"
+        },
+        {
+            "Address": [
+                "7",
+                "236",
+                "128",
+                "1",
+                "0",
+                "$bus % 16 * 5"
+            ],
+            "Bus": "$bus / 4 - 4",
+            "Class": "I3C",
+            "EndpointId": "$bus % 15 * 10 + 5",
+            "Name": "CXL2",
+            "Type": "MCTPEndpoint"
+        },
         {
             "CheckHysteresisWithSetpoint": true,
             "Class": "temp",
@@ -275,7 +320,7 @@
         }
     ],
     "Name": "Yosemite 4 Wailua Falls Slot $bus % 15",
-    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Wailua Falls', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4', 'BOARD_MANUFACTURER': '(Quanta|Wiwynn)'})",
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': '^Wailua Falls$', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4', 'BOARD_MANUFACTURER': '(Quanta|Wiwynn)'})",
     "Type": "Board",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
         "BuildDate": "$BOARD_MANUFACTURE_DATE",
@@ -298,4 +343,4 @@
         "Type": "ASIC"
     },
     "xyz.openbmc_project.Inventory.Item.Dimm": {}
-}
+}
\ No newline at end of file
diff --git a/configurations/meta/yv4/yosemite4_wailuafalls_without_asic.json b/configurations/meta/yv4/yosemite4_wailuafalls_without_asic.json
new file mode 100644
index 0000000..836a777
--- /dev/null
+++ b/configurations/meta/yv4/yosemite4_wailuafalls_without_asic.json
@@ -0,0 +1,102 @@
+{
+    "Exposes": [
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.035,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "WAILUA_FALLS_SLOT $bus % 15 WF_E1S_TEMP_C"
+            ],
+            "Name": "PID_WF_E1S_Slot $bus % 15",
+            "NegativeHysteresis": 3.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -5.5,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 69.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "WAILUA_FALLS_SLOT $bus % 15 WF_E1S_TEMP_C"
+            ],
+            "Name": "Stepwise_WF_E1S_Slot $bus % 15",
+            "NegativeHysteresis": 2,
+            "Output": [
+                10.0,
+                14.0,
+                18.0,
+                22.0,
+                26.0,
+                30.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                58.0,
+                59.0,
+                60.0,
+                61.0,
+                62.0,
+                63.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "InputUnavailableAsFailed": false,
+            "Inputs": [
+                "WAILUA_FALLS_SLOT $bus % 15 WF_INLET_TEMP_C"
+            ],
+            "MissingIsAcceptable": [
+                "WAILUA_FALLS_SLOT $bus % 15 WF_INLET_TEMP_C"
+            ],
+            "Name": "Stepwise_WF_INLET_Slot $bus % 15",
+            "NegativeHysteresis": 0,
+            "Output": [
+                10.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                10.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        }
+    ],
+    "Name": "Yosemite 4 Wailua Falls Slot $bus % 15",
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Wailua Falls without ASIC', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4', 'BOARD_MANUFACTURER': '(Quanta|Wiwynn)'})",
+    "Type": "Board",
+    "xyz.openbmc_project.Inventory.Decorator.Asset": {
+        "BuildDate": "$BOARD_MANUFACTURE_DATE",
+        "Manufacturer": "$BOARD_MANUFACTURER",
+        "Model": "$BOARD_PRODUCT_NAME",
+        "PartNumber": "$BOARD_PART_NUMBER",
+        "SerialNumber": "$BOARD_SERIAL_NUMBER",
+        "SparePartNumber": "$BOARD_INFO_AM1"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.AssetTag": {
+        "AssetTag": "$PRODUCT_ASSET_TAG"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.Revision": {
+        "Version": "$PRODUCT_VERSION"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.Slot": {
+        "SlotNumber": "$bus % 15"
+    }
+}
\ No newline at end of file
-- 
2.53.0-Meta

